lib/grpc/webproxy/GrpcWebProxyServer.ts

```diff
@@ -23,10 +25,15 @@ class GrpcWebProxyServer {
   /**
    * Start the server and begins listening on the specified proxy port.
    */
-  public listen = (proxyPort: number, grpcPort: number, grpcHost: string): Promise<void> => {
+  public listen = (proxyPort: number, grpcPort: number, grpcHost: string, tlsCertPath: string): Promise<void> => {
     // Load the proxy on / URL
     const protoPath = path.join(__dirname, '..', '..', '..', 'proto');
-    const gateway = grpcGateway(['xudrpc.proto'], `${grpcHost}:${grpcPort}`, grpc.credentials.createInsecure(), protoPath);
+    const gateway = grpcGateway(
+      ['xudrpc.proto'],
+      `${grpcHost}:${grpcPort}`,
+      grpc.credentials.createSsl(fs.readFileSync(tlsCertPath)),
```

Comment:
Right now we're reading this file twice, once when starting the grpc server and once for the web proxy. It would be nice if we only read it once, but no particularly clean way to do that jumps out at me. Maybe the `listen` method for GrpcServer could return the loaded cert, to then be used optionally for the web proxy? Not a big deal though so do you what you think is best.

Reply:
That way the gRPC server would take the path to the certificate as argument but the web proxy the certificate itself and I don't want to move the generation of the certificate into `Xud.ts` because it doesn't belong there.

I think that reading a file of the size of 800 bytes a second time on startup isn't that big of a overhead tbh.

Reply:
OK with me.